packages/orbit-components/src/Slider/index.tsx

```diff
@@ -69,403 +70,381 @@ const StyledSliderInput = styled.div`
-      this.setState({ value: newValue });
-    }
-  }
-
-  pauseEvent = (
-    event:
-      | React.KeyboardEvent<HTMLDivElement>
-      | React.MouseEvent<HTMLDivElement>
-      | React.TouchEvent<HTMLDivElement>
-      | React.FocusEvent<HTMLDivElement>,
-  ) => {
-    if (typeof event.stopPropagation === "function") {
-      event.stopPropagation();
-    }
-    if (
-      typeof event.preventDefault === "function" &&
-      (typeof event.cancelable !== "boolean" || event.cancelable)
-    ) {
-      event.preventDefault();
-    }
-  };
-
-  stopPropagation = (event: React.TouchEvent<HTMLDivElement>) => {
-    if (typeof event.stopPropagation === "function") event.stopPropagation();
-  };
-
-  isNotEqual = (a: Value, b: Value) => {
-    if (Array.isArray(a) && Array.isArray(b)) {
-      return a.toString() !== b.toString();
-    }
-    return a !== b;
-  };
-
-  calculateValue = (ratio: number, addition?: boolean, deduction?: boolean) => {
-    const { maxValue = DEFAULT_VALUES.MAX, minValue = DEFAULT_VALUES.MIN } = this.props;
-    return Math.round(
-      (maxValue - minValue + (addition ? 1 : 0)) * ratio + minValue - (deduction ? 1 : 0),
-    );
-  };
-
-  calculateValueFromPosition = (pageX: number, throughClick?: boolean) => {
-    // @ts-expect-error TODO
-    const barRect = boundingClientRect(this.bar);
-    if (barRect) {
-      const {
-        histogramData,
-        histogramLoading,
-        theme: { rtl },
-      } = this.props;
-      const { handleIndex, value } = this.state;
-      const mousePosition = (rtl ? barRect?.right : pageX) - (rtl ? pageX : barRect?.left);
-      const positionRatio = mousePosition / barRect.width;
-      const hasHistogram = histogramLoading || !!histogramData;
-      // when range slider
-      if (Array.isArray(value)) {
-        if (value[0] === value[value.length - 1]) {
-          if (this.calculateValue(positionRatio, true, true) >= value[value.length - 1]) {
-            return this.calculateValue(positionRatio, true, true);
+const PureSlider = ({
+  defaultValue = DEFAULT_VALUES.VALUE,
+  maxValue = DEFAULT_VALUES.MAX,
+  minValue = DEFAULT_VALUES.MIN,
+  step = DEFAULT_VALUES.STEP,
+  theme,
+  onChange,
+  onChangeAfter,
+  onChangeBefore,
+  ariaValueText,
+  ariaLabel,
+  label,
+  histogramData,
+  histogramLoading,
+  histogramDescription,
+  histogramLoadingText,
+  valueDescription,
+  id,
+  dataTest,
+}: SliderProps) => {
+  const bar = React.useRef<HTMLDivElement>(null);
+  const [value, setValue] = React.useState(defaultValue);
+  const handleIndex = React.useRef<number | null>(null);
+  const [focused, setFocused] = React.useState(false);
+  const { rtl } = theme;
+
+  React.useEffect(() => {
+    const newValue = Array.isArray(defaultValue)
+      ? defaultValue.map(item => Number(item))
+      : Number(defaultValue);
+
+    setValue(newValue);
+  }, [defaultValue]);
+
+  const calculateValue = React.useCallback(
```

Comment:
I think you are overusing the useCallbacks here, profile first before those optimizations, they seem as not needed to be here everywhere

Reply:
Agreed. I ended up abstracting a lot of the functions to the utils file, so that they are just called and not redefined on every render. The only useCallback usage is for props that are not expected to change often and, therefore, can have a callback optimized.